proposals/0000-improved-ghc-development.rst

```diff
@@ -0,0 +1,216 @@
+
+The often proposed solution is to move everything to GitHub. Some of us believe GitHub is not up to the task of managing a project such as GHC. But we also believe with minor changes we can greater leverage the user base of GitHub while also providing valuable information back to the non GitHub users.
+
+Proposed Change
+---------------
+- Link Phabricator and GitHub
+
+  Phabricator seems to have built in support for OAuth. One easy way so lower the barrier is to support OAuth logins to GitHub. This way a user doesn't have to create a new account. They can simply login using their GitHub credentials. *(This seems to already be enabled, maybe we should advertise it more.)*
+
+  Bonus points if the GitHub API allows us to extract the user's public key from GitHub. This would also mean less setup for the user to be able to push a diff.
+
+  Phabricator documentation for this https://secure.phabricator.com/book/phabricator/article/configuring_accounts_and_registration/
+  
+- Link Trac to GitHub
+
+  We already have a read-only mirror for code that is incredibly useful. I propose extending this read-only mode further. 
+
+   * Support OAuth on Trac for logins against GitHub as well.
+   * Synchronize Trac issues in a read-only way to GitHub. The usefulness of this feature comes In the ability for users to reference the issues from their own GitHub projects. This gives us some valuable information such as who these issues affect. This information would get synched back to Trac. This also allows users to be notified when something gets fixed. Since the issue would be closed on GitHub as well. Finally it gives some information on the importance of issues.
+
+  With the ability to login directly on Trac using your GitHub account, I believe it's easier to take issues on one place: Trac. So read-only would be enough.
+
+  As a whole, I find Trac MUCH better for ticket triaging than Github or Phab,
+  both of which seem to be quite bare and simple in what they provide. I am not
+  in favor of ditching it. Whatever we gain we lose in the ability to generate custom views on issues. Which is a powerful filtering mechanism for prioritization.
+
+  Also we have and continue to accept patches just uploaded
+  to Trac as a diff. We tend to ask people to upload it to phab for better reviews
+  and so it's attributed to them when we commit. Some don't (and we then do it ourselves),
+  most due. If they don't need another login then I suspect almost all would.
```

Comment:
`s/due/do/`.

Indeed; I don't think much needs to change here. Handling the occasional Trac patch doesn't take *that* much time assuming it applies cleanly.